src/Zap.sol

```diff
@@ -370,13 +409,16 @@ contract Zap is Errors {
             params: params,
             referralCode: REFERRAL_CODE
         });
+
+        stage = Stage.UNSET;
```

Comment:
won't `stage` be `UNSET` (and not `LEVEL1`) at execution time?

Reply:
1. `unwind()` confirms the `stage` is `UNSET` and sets it to `LEVEL1`
2. `flashLoanSimple()` is called, which itself will call (i.e., callback) `executeOperation()`, thus passing context to `executeOperation()`
3. `executeOperation()` confirms the `stage` is `LEVEL1` and sets it to `LEVEL2`
4. After `executeOperation()` returns context to `flashLoanSimple()`, it returns context to `unwind()`, which finally sets stage back to `UNSET`

Reply:
e2e test will confirm this though